##### Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170. alpha = 0.05.

Соберем в массивы значения роста, считаем их количество в каждой группе и общее количество:

In [2]:
import numpy as np

In [4]:
y1 = np.array([173, 175, 180, 178, 177, 185, 183, 182]) # футболисты
y2 = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180]) # хоккеисты
y3 = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170]) # штангисты
n1, n2, n3, k = len(y1), len(y2), len(y3), 3
n = n1 + n2 + n3
print(f'футболистов: {n1}\nхоккеистов: {n2}\nштангистов: {n3}\nвсего человек: {n}\nгрупп: {k}')

футболистов: 8
хоккеистов: 9
штангистов: 11
всего человек: 28
групп: 3


Найдем средний рост для каждой группы и для всех вместе:

In [5]:
y1_mean, y2_mean, y3_mean, y_all = np.mean(y1), np.mean(y2), np.mean(y3), np.concatenate([y1, y2, y3])
y_mean = np.mean(y_all)
print(f'средний рост футболистов: {np.round(y1_mean, 2)}\n'
      f'средний рост хоккеистов: {np.round(y2_mean, 2)}\n'
      f'средний рост штангистов: {np.round(y3_mean, 2)}\n'
      f'средний рост всех: {np.round(y_mean, 2)}')

средний рост футболистов: 179.12
средний рост хоккеистов: 178.67
средний рост штангистов: 172.73
средний рост всех: 176.46


Найдем три разных суммы квардатов отклонений:

In [6]:
s2 = np.sum((y_all - y_mean)**2)
s2_f = ((y1_mean - y_mean)**2) * n1 + ((y2_mean - y_mean)**2) * n2 + ((y3_mean - y_mean)**2) * n3
s2_residual = s2 - s2_f
print(f's2 - сумма квадратов отклонений наблюдений от общего среднего = {np.round(s2, 2)}\n'
      f's2_f - сумма квадратов отклонений средних групповых значений от общего среднего = {np.round(s2_f, 2)}\n'
      f's2_residual - остаточная сумма квадратов отклонений = {np.round(s2_residual, 2)}')

s2 - сумма квадратов отклонений наблюдений от общего среднего = 830.96
s2_f - сумма квадратов отклонений средних групповых значений от общего среднего = 253.91
s2_residual - остаточная сумма квадратов отклонений = 577.06


Проверка:

In [7]:
'Верно' if s2 == s2_f + s2_residual else 'Ошибка'

'Верно'

Находим 3 разных дисперсии:

In [8]:
sigma2_general = s2 / (n - 1)
sigma2_f = s2_f / (k - 1)
sigma2_residual = s2_residual / (n - k)
print(f'sigma2_general - общая дисперсия = {np.round(sigma2_general, 2)}\n'
      f'sigma2_f - факторная дисперсия = {np.round(sigma2_f, 2)}\n'
      f'sigma2_residual - остаточная дисперсия = {np.round(sigma2_residual, 2)}')

sigma2_general - общая дисперсия = 30.78
sigma2_f - факторная дисперсия = 126.95
sigma2_residual - остаточная дисперсия = 23.08


Найдём расчётное значение $F_{H}$ и табличное $F_{крит}$ в таблице критических точек распределения Фишера-Снедекора с характеристиками:

Заданный уровень значимости $\alpha = 0.05$

Cтепени свободы:
$$df_{межд} = k - 1 = 3 - 1 = 2$$
$$df_{внутр} = n - k = 28 - 3 = 25$$
$$\Rightarrow F_{крит} = 3.38$$


In [9]:
F_h = sigma2_f / sigma2_residual
print(f'F_h = {np.round(F_h, 2)}')

F_h = 5.5


Найдем эмпирическое корреляционное отношение $η^2$:

In [10]:
eta2 = s2_f / s2
print(f'eta2 = {np.round(eta2, 2)}')

eta2 = 0.31


$F_{H} = 5.5 > F_{крит} = 3.38 \Rightarrow$ различие средних значений роста в трех группах статистически значимо

$η^2 = 0.31 \Rightarrow$ для более корректного исследования требуется более глубокий анализ